File: examples/async/bybit_ws_public.py
"""Read Bybit public WebSocket trade events."""

import asyncio

from dcex.ws import bybit


async def main() -> None:
    """Subscribe to Bybit spot trades and print two messages."""
    async with bybit.public(category="spot") as ws:
        await ws.subscribe_trades("BTC-USDT-SPOT")
        print(await ws.recv())
        print(await ws.recv())


if __name__ == "__main__":
    asyncio.run(main())

File: crates/dcex/examples/bybit_ws_public.rs
use std::time::Duration;

use dcex::ws::bybit::BybitPublicWebSocket;

#[tokio::main]
async fn main() -> dcex::Result<()> {
    let mut ws = BybitPublicWebSocket::new("spot", Duration::from_secs(10))?;
    ws.connect().await?;
    ws.subscribe_trades("BTC-USDT-SPOT").await?;
    println!("{}", ws.recv().await?);
    println!("{}", ws.recv().await?);
    ws.close().await?;
    Ok(())
}
